src/algebra/big_operators.lean

```diff
@@ -462,6 +462,12 @@ lemma prod_range_succ' (f : ℕ → β) :
 | (n + 1) := by rw [prod_range_succ (λ m, f (nat.succ m)), mul_assoc, ← prod_range_succ'];
                  exact prod_range_succ _ _
 
+lemma le_sum_nat_of_mem (f : ℕ → ℕ) (s : finset ℕ) (a : ℕ) (t : a ∈ s) : f a ≤ ∑ i in s, f i :=
+begin
+  rw ← finset.insert_erase t,
+  simp only [sum_insert, eq_self_iff_true, mem_erase, not_true, zero_le, le_add_iff_nonneg_right, ne.def, not_false_iff, false_and],
```

Comment:
Probably someone encouraged you along the way to replace non-terminal `simp`s with the output of `squeeze_simp`.

In the other direction, I think opinions vary about whether it's a good idea to replace terminal `simp`s. The argument in favour is that it marginally speeds things up (much less so since Lean 3.13). The argument against is that it's a visual mess, and arguably harder to maintain (e.g. if simp lemmas get renamed).